_posts/2021-06-23-devfile-v2-and-ide-plugins.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Devfile v2 and IDE plug-ins
+layout: post
+author: Florent Benoit
+description: >-
+  How are handle IDE plug-ins with devfile v2
+categories: []
+keywords: ['devfile', 'ide', 'plug-ins']
+slug: /@florent.benoit/devfile-v2-and-ide-plug-ins
+---
+
+### Definition
+
+Devfile has been incubated by Eclipse Che project providing [devfile v1 specification](https://redhat-developer.github.io/devfile/).
+
+To allow other projects to use a common definition, devfile has moved out of Eclipse Che. It has its own definition at [https://devfile.io](https://devfile.io) with current [v2.1.0 specification](https://docs.devfile.io/devfile/2.1.0/user-guide/api-reference.html).
+
+Devfile v1 were handled by Eclipse Che Server workspace engine using Java/Kubernetes API. There is a new engine for devfile v2 called DevWorkspace Controller. The source code is at: [https://github.com/devfile/devworkspace-operator](https://github.com/devfile/devworkspace-operator).
+
+DevWorkspaces are custom kubernetes resources and can be created/listed/deleted using kubernetes tools like `kubectl`.
+
+A devWorkspace is a workspace using the new engine and supporting devfile V2 definition.
+
+### Milestones
+
+Devfile v2 support in Eclipse Che can be followed by milestones.
+
+*   [Milestone 1](https://github.com/eclipse/che/milestone/136) introduced the first support of devfile v2 (start a basic workspace)
+*   [Milestone 2](https://github.com/eclipse/che/milestone/139) just landed in Eclipse Che 7.32.0 with the support of Theia plug-ins for DevWorkspaces
+*   [Milestone 3](https://github.com/eclipse/che/milestone/140) is the next target, including Kubernetes support and more.
+
+### How to test ?
+
+The support of devfile v2/devWorkspaces is optional and need to be opt-in.  
+
+When the support is enabled, both devfile v1 and v2 are supported. You can still work with your previous workspaces and devfile v1.
+
+* If the `devfile.yaml` has `apiVersion 1.0.0`, it will use che server workspace engine to deploy this devfile.
+* If the `devfile.yaml` has `schemaVersion 2.0.0` (or greater like `2.1.0`), it will use the devWorkspace controller to deploy this devfile.
+
+⚠️ DevWorkspaces requires OpenShift as platform until Milestone 3 is reached (no support of plain Kubernetes yet)
+
+ℹ️ Enabling devWorkspace is also defaulting installation to use single-host deployment. So no need to install certificates on your computer.
+
+To enable devWorkspace, it uses a custom spec object for Che Operator:  
```

Comment:
maybe it would make sense to capitalize `devWorkspace` for consistency? I noticed that in the blog post both variants are mixed  devWorkspace / DevWorkspaces

Reply:
I think I used capital letter when starting a new sentence

Reply:
I'll switch to `DevWorkspace` everywhere